package.json

```diff
@@ -60,17 +60,17 @@
     "screenfull": "^5.1.0",
     "seedrandom": "^3.0.5",
     "shutterbug": "^1.3.2",
-    "three": "^0.119.1",
+    "three": "^0.130.0",
     "timeseries-analysis": "^1.0.12",
     "ts-migrate": "^0.1.19",
     "uuid": "^8.3.2"
   },
   "devDependencies": {
     "@svgr/webpack": "^5.5.0",
-    "@types/d3-color": "^2.0.1",
+    "@types/d3-color": "^3.0.1",
     "@types/d3-hsv": "^0.1.4",
-    "@types/d3-interpolate": "^2.0.0",
-    "@types/d3-scale": "^3.2.2",
+    "@types/d3-interpolate": "^3.0.0",
+    "@types/d3-scale": "^4.0.0",
```

Comment:
Generally speaking, `@types` major version numbers track the corresponding library's major version numbers. If you're sticking with `d3-color: ^2.0.0`, for instance, you should stick with `@types/d3-color: ^2.0.0`, otherwise you run the risk of having type definitions that don't correspond to the version of the code you're using. As for the underlying issue, I've run into the published-as-ESM-modules problem with d3 and other libraries as well, and agree that it's annoying.

Reply:
Thanks, I've forgotten to revert versions of @types packages. I've just fixed that in the last commit.


Re d3, ESM and other breaking changes - I cannot count the number of times D3 libs did that to me. It's definitely the lib that I wasted the most time on trying to update versions or make them work in a new environment (ofc not in this project). With all the respect to its developers, maintainers, and their good ideas - no idea why I keep using them. :wink:

Reply:
I've also run into it with a number of Sindre Sorhus' libraries, so d3 aren't the only ones.